### ISSUE TITLE
MultipartRequestHelper.cs  in  Docs\aspnetcore\mvc\models\file-uploads\sample\FileUploadSample\    is throwing type errors on  netcore 2.0

### ISSUE THREAD

**Issue by EdAFranco** [joeaudette/cloudscribe#306]
It seems ContentDispositionHeaderValue.cs  which is part of the Microsoft.Net.Http.Headers namespace  was updated in netcore 2.0 to:

(netcore1.0 left - netcore 2.0 right)

![image](https://user-images.githubusercontent.com/34019776/33253528-f1dce04c-d397-11e7-8c09-d7dec275fdfe.png)

This change impacts MultipartRequestHelper.cs  and throws:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1503	Argument 1: cannot convert from 'Microsoft.Extensions.Primitives.StringSegment' to 'string'	App1	C:\...\MultipartRequestHelper.cs
```

How to modify MultipartRequestHelper.cs  so it runs with Assembly Microsoft.Net.Http.Headers, Version=2.0.1.0,?

Thank you.

**Comment by EdAFranco** [comment]
I believe I fixed it just by using the .Value  property  described here: [](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringsegment.value?view=aspnetcore-2.0#Microsoft_Extensions_Primitives_StringSegment_Value)

**Comment by joeaudette** [comment]
Hi, That class is currently not even used in cloudscribe.FileManager.Web. Nothing references it and there are no errors or warnings building it against 2.0 asp.net. That was some code I copied from docs but did not end up using it anywhere yet. Since the code currently calls .ToString on boundary that would convert the StringSegment into string anyway, so no code change is needed as far as I know.

That code is for streaming large files, I found it from the documentation here: https://docs.microsoft.com/en-us/aspnet/core/mvc/models/file-uploads

At the top they link to this github repo and that is where the code for MultiPartRequestHelper.cs came from: https://github.com/aspnet/Docs/tree/master/aspnetcore/mvc/models/file-uploads/sample/FileUploadSample

As mentioned, I am not even using that code anywhere in cloudscribe, it was included for future use to handle streaming of very large file uploads.

The original file is here: https://github.com/aspnet/Docs/blob/master/aspnetcore/mvc/models/file-uploads/sample/FileUploadSample/MultipartRequestHelper.cs

but my copy is already calling .ToString() on boundary which is probably a change I made to update it to compile against 2.0.
